test/msw-api/setup-worker/use.test.ts

```diff
@@ -108,3 +109,45 @@ test('returns a mocked response from a one-time request handler override only up
 
   await runtime.cleanup()
 })
+
+test('returns a mocked response from a one-time request handler override only upon first request match with parallel requests', async () => {
+  const runtime = await runBrowserWith(path.resolve(__dirname, 'use.mocks.ts'))
+
+  await runtime.page.evaluate(() => {
+    const { msw } = window
+
+    msw.worker.use(
+      msw.rest.get('/book/:bookId', function oneTimeOverride(req, res, ctx) {
+        return res.once(ctx.json({ title: 'One-time override' }))
+      }),
+    )
+  })
```

Comment:
@timdeschryver not sure if I got the suggestion, would it be that:

```suggestion
  msw.worker.use(
    msw.rest.get('/book/:bookId', function oneTimeOverride(req, res, ctx) {
      const { bookId } = req.params
      return res.once(
        ctx.json({ title: `One-time override; bookId = '${bookId}'` }),
      )
    }),
  )
```

And then an assertion like that:

```ts
expect(bookBody).toEqual({ title: `One-time override; bookId = 'abc-123'` })
```

Reply:
You can just add a `bookId` property on the response JSON body directly:

```js
return res.once(ctx.json({ title: '...', bookId }))
```

No need to put all that data into a single string. You can later ensure that the right one-time handler has resolved by adding the `bookId` to the expected object:

```js
fetch('/book/abc-123')
fetch('/book/def-456')

// Checking the `bookId` value so we're sure which response we got. 
expect(bookBody).toEqual({ title: 'One-time override', bookId: 'abc-123' })
```

Reply:
Got it, will give it a try!

Reply:
Ok, so I've added the `bookId` to the response body as suggested. It looks good!

I've got two related questions:

1. Should I do the same for the one-time handler of the other existing tests (the ones with sequential execution) so the tests are more consistent? Or it's out of the scope of this PR?
2. Should I add the `bookId` param to the default handler's body as well? Or it's better to keep it only in the one-time handler, so they have two differences between each other (the `title` and the `bookId` param)?

Reply:
Thank you for adding that!

1. No, I think there's no need. Unless those tests are related to the issue we're fixing, we'd better leave them as-is.
2. Let it be only in the one-time handler. The default handler won't have the `bodyId`, which would fail the assertion in case of an issue. That's what we want. At this point there is no difference if the assertion fails because the key is missing, or because it has a wrong value.

Reply:
Sounds good, I've updated it based on the above, please have a look!